src/Aspire.Hosting/ApplicationModel/ContainerImageReference.cs

```diff
@@ -21,4 +23,13 @@ public class ContainerImageReference(IResource resource) : IManifestExpressionPr
 
     /// <inheritdoc/>
     public IEnumerable<object> References => [Resource];
+
+    /// <inheritdoc/>
+    async ValueTask<string?> IValueProvider.GetValueAsync(CancellationToken cancellationToken)
+    {
+        var deploymentTarget = Resource.GetDeploymentTargetAnnotation() ?? throw new InvalidOperationException($"Resource '{Resource.Name}' does not have a deployment target.");
+        var containerRegistry = deploymentTarget.ContainerRegistry ?? throw new InvalidOperationException($"Resource '{Resource.Name}' does not have a container registry.");
+        var registryEndpoint = await containerRegistry.Endpoint.GetValueAsync(default).ConfigureAwait(false);
+        return $"{registryEndpoint}/{Resource.Name.ToLowerInvariant()}:latest";
```

Comment:
Open an issue about this an ContainerPortReference. We need to resolve it from the output of the build.

Reply:
See https://github.com/dotnet/aspire/issues/11033.